apps/onboarding/lib/view-model.js

```
@@ -247,32 +252,39 @@ export const submitClassFlow = async state => {
       }
       `
     )
-    const { createAppointment: appointment } = await gql.request(
-      `
-      mutation {
-        createAppointment(
-          appointmentId: "${state.selectedAvailability.id}"
-          appointmentTypeId: "${state.apptTypeId}"
-          startTime: "${state.selectedAvailability.startTime}"
-          leadEmail: "${lead.email}"
-        ) {
-          id
-          endTime
-          startTime
-          location {
-            id
-            address1
-            city
-            state
-            zipCode
-            country
+    if (lead.id) {
```

I don't think we want to wrap this in an `if`. If we fail to return a lead from `createLead` then we want to just let the function fail loudly instead of silently succeeding and not actually creating the client or appointment.